src/commonMain/kotlin/sim/rotation/criteria/ResourcePctLte.kt

```diff
@@ -34,7 +30,8 @@ class ResourcePctLte(data: RotationRuleCriterion) : Criterion(Type.RESOURCE_PCT_
     }
 
     override fun satisfied(sp: SimParticipant): Boolean {
-        if(pct == null || resourceType == null) {return false}
-        return sp.getResource(resourceType).currentAmount <= sp.getResource(resourceType).maxAmount * (pct / 100.0)
+        val res = sp.resources[resourceType]
+        if(pct == null || resourceType == null || res == null) {return false}
```

Comment:
Same here, print a warning or no?

Reply:
Given that criteria consume user input, we should probably warn and return false - the most likely cause is a typo in a rotation.